src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java

```diff
@@ -172,20 +175,48 @@ <T> Stream<T> getDependencies(
       return function.apply(originalConfiguration.getResolvedConfiguration());
     }
     catch (ResolveException e) {
-      Configuration copyConfiguration = createCopyConfiguration(project);
+      Configuration copyConfiguration = copyDependencies(project, originalConfiguration, false);
 
-      originalConfiguration.getAllDependencies().all(dependency -> {
-        if (dependency instanceof ProjectDependency) {
-          Project dependencyProject = ((ProjectDependency) dependency).getDependencyProject();
-          project.evaluationDependsOn(dependencyProject.getPath());
-        }
-        else {
-          copyConfiguration.getDependencies().add(dependency);
+      try {
+        return function.apply(copyConfiguration.getResolvedConfiguration());
+      }
+      catch (Exception lastResortException) {
+        copyConfiguration = copyDependencies(project, originalConfiguration, true);
+        return function.apply(copyConfiguration.getResolvedConfiguration());
```

Comment:
I'm just wondering how this third check  will effect performance (if at all) ? :thinking: Other builds should resolve as normal right? Seems this scenario is very specific e.g. an android build using dependency substitution rules.  I suppose these builds might take a little longer but at least they complete (and skipped dependencies are logged)

Reply:
> I'm just wondering how this third check will effect performance? 🤔 Other builds should resolve as normal right? Seems this scenario is very specific e.g. an android build using dependency substitution rules. I suppose these builds might take a little longer but at least they complete (and skipped dependencies are logged)

Right, this is only happening for Android projects and would only be executed if the first attempt fails.
At the end, any extra delay would be preferred to a failure at this point.